examples/ex-layout.cpp

```diff
@@ -0,0 +1,197 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-18, Lawrence Livermore National Security, LLC.
+//
+// Produced at the Lawrence Livermore National Laboratory
+//
+// LLNL-CODE-689114
+//
+// All rights reserved.
+//
+// This file is part of RAJA.
+//
+// For details about use and distribution, please read RAJA/LICENSE.
+//
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include <cstdlib>
+#include <cstring>
+#include <iostream>
+
+#include "RAJA/RAJA.hpp"
+#include "memoryManager.hpp"
+
+/*
+ *  Layout Example
+ *
+ *  Populates a 4-dimensional array
+ *
+ *  RAJA features shown:
+ *    - `forall` loop iteration template method
+ *    -  basic usage of the layout object
+ */
+
+//
+// Function to check result
+//
+template<typename T>
+void checkResult(T A, T B, int len);
+
+int main(int RAJA_UNUSED_ARG(argc), char **RAJA_UNUSED_ARG(argv[]))
+{
+
+//
+//Define dimension
+//
+  const int dim = 4;
+
+//
+// Define stride length
+//  
+  const int Istride = 2;
+  const int Jstride = 3;
+  const int Mstride = 4;
+  const int Nstride = 5;
+
+//
+//Define range segment
+//
+  RAJA::RangeSegment Irange(0,Istride);
+  RAJA::RangeSegment Jrange(0,Jstride);
+  RAJA::RangeSegment Mrange(0,Mstride);
+  RAJA::RangeSegment Nrange(0,Nstride);
+
+//
+// Allocate 1D arrays
+//
+  int array_len = Istride*Jstride*Mstride*Nstride;
+
+  int* A = new int[array_len];
+  int* B = new int[array_len];
+
+  //Wrap A in a view
+  RAJA::View<int, RAJA::Layout<dim> > A_IJMN_view(A, Istride, Jstride, Mstride, Nstride);
+
+  //Create layout objects assuming an IJMN indexing
+  RAJA::Layout<dim> IJMN_layout = RAJA::make_permuted_layout({Istride,Jstride,Mstride,Nstride}, RAJA::as_array<RAJA::Perm<0,1,2,3>>::get());
+  //Equivalent to : RAJA::Layout<dim> IJMN_Layout(Istride,JStride,MStride,Nstride);
+
+//----------------------------------------------------------------------------//
+//
+// In the following example we show how a layout
+// object may be used to convert a 1D index to a 4D index.
```

Comment:
I don't think the layout -> indices conversion is so useful inside of a loop. There should be a better way to highlight this.